notebooks/data_loader.ipynb

```
@@ -0,0 +1,147 @@
+    "def get_standardized_data(cache_path=DEFAULT_CACHE_PATH, force_download=False):\n",
+    "    # 1. Check Cache\n",
+    "    if os.path.exists(cache_path) and not force_download:\n",
+    "        logger.info(f\"Loading standardized data from cache: {cache_path}\")\n",
+    "        return pd.read_csv(cache_path, index_col='datetime', parse_dates=True)\n",
+    "\n",
+    "    # 2. Fetch Data\n",
+    "    logger.info(\"Fetching dataset from UCI Repository...\")\n",
+    "    try:\n",
+    "        data = fetch_ucirepo(id=381)\n",
+    "        df = pd.concat([data.data.features, data.data.targets], axis=1)\n",
+    "    except Exception as e:\n",
```

Ruff (0.15.10)

[error] Undefined name `display`

(F821)